src/main/java/org/jabref/logic/openoffice/action/EditInsert.java

```diff
@@ -0,0 +1,115 @@
+import org.jabref.model.openoffice.uno.UnoScreenRefresh;
+import org.jabref.model.openoffice.util.OOListUtil;
+
+import com.sun.star.beans.IllegalTypeException;
+import com.sun.star.beans.NotRemoveableException;
+import com.sun.star.beans.PropertyVetoException;
+import com.sun.star.lang.WrappedTargetException;
+import com.sun.star.text.XTextCursor;
+import com.sun.star.text.XTextDocument;
+
+public class EditInsert {
+
+    private EditInsert() {
+        /**/
+    }
+
+    /**
+     * In insertEntry we receive BibEntry values from the GUI.
+     *
+     * In the document we store citations by their citation key.
+     *
+     * If the citation key is missing, the best we can do is to notify the user. Or the programmer,
+     * that we cannot accept such input.
+     *
+     */
+    private static String insertEntryGetCitationKey(BibEntry entry) {
+        Optional<String> key = entry.getCitationKey();
+        if (key.isEmpty()) {
+            throw new IllegalArgumentException("insertEntryGetCitationKey: cannot cite entries without citation key");
+        }
+        return key.get();
+    }
+
+    /**
+     * @param cursor Where to insert.
+     * @param pageInfo A single pageInfo for a list of entries. This is what we get from the GUI.
+     */
+    public static void insertCitationGroup(XTextDocument doc,
+                                           OOFrontend frontend,
+                                           XTextCursor cursor,
+                                           List<BibEntry> entries,
+                                           BibDatabase database,
+                                           OOBibStyle style,
+                                           CitationType citationType,
+                                           String pageInfo)
+        throws
+        NoDocumentException,
+        NotRemoveableException,
+        WrappedTargetException,
+        PropertyVetoException,
+        CreationException,
+        IllegalTypeException {
+
+        List<String> citationKeys = OOListUtil.map(entries, EditInsert::insertEntryGetCitationKey);
+
+        final int nEntries = entries.size();
+        List<Optional<OOText>> pageInfos = OODataModel.fakePageInfos(pageInfo, nEntries);
+
+        List<CitationMarkerEntry> citations = new ArrayList<>(nEntries);
+        for (int i = 0; i < nEntries; i++) {
+            Citation cit = new Citation(citationKeys.get(i));
+            cit.lookupInDatabases(Collections.singletonList(database));
+            cit.setPageInfo(pageInfos.get(i));
+            citations.add(cit);
+        }
+
+        // The text we insert
+        OOText citeText =
+            (style.isNumberEntries()
+             ? OOText.fromString("[-]") // A dash only. Only refresh later.
+             : style.createCitationMarker(citations,
+                                          citationType.inParenthesis(),
+                                          NonUniqueCitationMarker.FORGIVEN));
```

Comment:
On more complex if clauses, please avoid `? :` for better readability

Reply:
Changed to
```{java}
        OOText citeText = null;
        if (style.isNumberEntries()) {
            citeText = OOText.fromString("[-]"); // A dash only. Only refresh later.
        } else {
            citeText = style.createCitationMarker(citations,
                                                  citationType.inParenthesis(),
                                                  NonUniqueCitationMarker.FORGIVEN);
        }
```